src/kixi/nhs/board_report.clj

```diff
@@ -23,35 +24,145 @@
                 (assoc :indicator_id indicator-id))) data)))
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
-;; Internal calculations                                                                ;;
+;; Internal calculations - indicator: 212                                               ;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(defn split-by-key
+  "Turns a sequence of maps into a sequence of sequences,
+  where each nested sequence corresponds to a single group."
+  [k data]
+  (->> (group-by k data)
+       vals))
+
+(defn get-slope [y]
+  (let [{:keys [year period_of_coverage]} (first y)
+        x [0.1 0.3 0.5 0.7 0.9]
+        [_ slope] (:coefs (stats/linear-model (map :indicator_value y) x))]
+    {:year year
+     :period_of_coverage period_of_coverage
+     :slope slope}))
+
+(defn average
+  "Calculates average of a collection."
+  [coll]
+  (float (/ (reduce + coll) (count coll))))
+
+(defn deprivation-groups-avg
+  "Average the deprivation values two by two so that you end up with 5 deprivation groups.
+  i.e. average the values for deprivation groups 1 and 2 to get a new value for a new
+  deprivation group 1. Then average the values for deprivation groups 3 and 4 to get a valu
+  e for a new deprivation group 2 etc until averaging values for deprivation groups 9 and 10
+  to get a value for a new deprivation group 5. then proceed with the steps."
+  [data]
+  (->> (remove #(= (:level %) "Unknown") data)
+       (map #(update-in % [:level] transform/parse-number))
+       (sort-by :level)
+       (partition 2)
+       (map-indexed (fn [idx level-partition]
+                      {:level (inc idx)
+                       :period_of_coverage (-> level-partition first :period_of_coverage)
+                       :year (-> level-partition first :year)
+                       :indicator_value (average (->> (map :indicator_value level-partition)
+                                                      (map transform/parse-number)))}))))
+(defn get-year-x
+  "Filter value for the latest date minus x years."
+  [x data]
+  (let [sorted (sort-by :year data)]
+    (if (< x (count data))
+      (->> sorted (drop-last x) last)
+      (first sorted))))
+
+(defn get-median-indicators
+  "Returns a sequence of indicators for deprivation group 3
+  for all years."
+  [data]
+  (keep #(when (= (:level %) 3)
+           {:year (:year %)
+            :period_of_coverage (:period_of_coverage %)
+            :median_indicator_value (:indicator_value %)}) data))
+
+(defn divide-slope-by-median
+  "Divide the slope value found in step 2 by the indicator value of deprivation group 3."
+  [data]
+  (assoc data :division (float (/ (:slope data) (:median_indicator_value data)))))
+
+(defn deprivation-year-1
+  "Calculate the difference between the latest value calculated on step 3
+  and the same value from § years previously to the latest year.
```

Comment:
I copied it from the recipe, don't kill the messenger please.
I assumed it means `(- last-year 1)` and `(- last-year 5)`.